diff --git a/Services/FeedService.php b/Services/FeedService.php
index cb13482..0133cc8 100644
--- a/Services/FeedService.php
+++ b/Services/FeedService.php
@@ -51,12 +51,12 @@ class FeedService {
     foreach ($slides as $slide) {
       $options = $slide->getOptions();
 
-      $source = $options['source'];
-
-      if (empty($source)) {
+      if (empty($options['source'])) {
         continue;
       }
 
+      $source = $options['source'];
+
       $md5Source = md5($source);
 
       // Check for previouslyDownloaded feed.
@@ -68,11 +68,11 @@ class FeedService {
       }
       else {
         // Fetch the FeedReader
-        $reader = $this->container->get('debril.reader');
+        $reader = $this->container->get('feedio');
 
         try {
           // Fetch content
-          $feed = $reader->getFeedContent($source);
+          $feed = $reader->read($source)->getFeed();
 
           // Setup return array.
           $res = array(
@@ -83,12 +83,11 @@ class FeedService {
           );
 
           // Get all items.
-          $items = $feed->getItems();
-
-          foreach ($items as $item) {
+          /** @var \FeedIo\Feed\Item $item */
+          foreach ($feed as $item) {
             $res[0]['feed'][] = array(
               'title' => $item->getTitle(),
-              'date' => $item->getUpdated()->format('U'),
+              'date' => $item->getLastModified()->format('U'),
               'description' => $item->getDescription(),
             );
           }
